source/upgrade.txt

```diff
@@ -0,0 +1,125 @@
+.. _csharp-upgrade-driver:
+
+=======================
+Upgrade Driver Versions
+=======================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+
+Overview
+--------
+
+In this section, you can identify essential changes you must make to your
+application when you upgrade your driver to a new version.
+
+Before you upgrade, perform the following actions:
+
+- Ensure the new version is compatible with the MongoDB Server versions
+  your application connects to and the {+framework+} version your
+  application runs on. See the :ref:`.NET/C# Driver Compatibility <csharp-compatibility-tables>`
+  page for this information.
+- Address any breaking changes between the current version of the driver
+  your application is using and your planned upgrade version in the
+  :ref:`Breaking Changes <csharp-breaking-changes>` section. To learn
+  more about the MongoDB Server release compatibility changes, see the
+  :ref:`<csharp-server-release-changes>` section.
+
+.. tip::
+
+   To minimize the amount of changes your application may require when
+   upgrading driver versions in the future, use the
+   :ref:`{+stable-api+} <csharp-stable-api>`.
+
+.. _csharp-breaking-changes:
+
+Breaking Changes
+----------------
+
+A breaking change is a modification in a convention or behavior starting in
+a specific version of the driver. This type of change may prevent your application from
+working properly if not addressed before upgrading the driver.
+
+The breaking changes in this section are categorized by the driver version that
+introduced them. When upgrading driver versions, address all the breaking
+changes between the current and upgrade versions. For example, if you
+are upgrading the driver from v2.0 to v2.20, address all breaking changes from
+the version after v2.0 including any listed under v2.20.
+
+Version 2.19.0 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- The ``ObjectSerializer`` only allows deserialization of
+  types that are considered safe. Previously, it allowed deserialization
+  of all types. What types are considered safe is
+  determined by the configurable ``AllowedTypes`` function (of type
+  ``Func<Type, bool>``). The default ``AllowedTypes`` function is
+  ``ObjectSerializer.DefaultAllowedTypes`` which returns "true" for a
+  number of well-known framework types that we have deemed safe.
+  A typical example might be to allow all the default allowed types as
+  well as your own types.
+
+  More information about the ``ObjectSerializer`` can be found in our
+  :ref:`FAQ <csharp-faq-object-serializer>`.
```

Comment:
Suggestion:

I think the style guide suggests [avoiding passive voice](https://www.mongodb.com/docs/meta/style-guide/writing/use-active-voice/) when possible.
The style guide also provides guidance on introducing links to clarify the info about them in 
[Make Link Text Clear](https://www.mongodb.com/docs/meta/style-guide/style/links-and-cross-references/#make-link-text-clear).

If you look through our docs, we usually use "To learn more, see..." or "For more information, see..." which are used in this PR a few times.

Reply:
Thank you, this is very useful!